sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -112,6 +112,24 @@ object GpuSemaphore {
     }
   }
 
+  def forbidVoluntaryRelease(context: TaskContext): Unit= {
+    if (context != null) {
+      getInstance.forbidVoluntaryRelease(context)
+    }
+  }
+
+  def allowVoluntaryRelease(context: TaskContext): Unit = {
+    if (context != null) {
+      getInstance.allowVoluntaryRelease(context)
+    }
+  }
+
+  def voluntaryRelease(context: TaskContext): Unit = {
```

Comment:
Can we please have some docs here to explain what is going on? This feels like a totally different feature from hash re-partitioning, and I would like to measure performance changes for it separate from anything with hash aggregation changes.

Reply:
This piece of code is irrelavant to repartition based agg, so it will be removed by a subsequent commit